bndtools.builder/src/org/bndtools/builder/classpath/BndContainerInitializer.java

```diff
@@ -61,15 +64,25 @@
 
     public BndContainerInitializer() {
         super();
-        Central.getInstance().addModelListener(this);
     }
 
     @Override
-    public void initialize(IPath containerPath, IJavaProject javaProject) throws CoreException {
-        IProject project = javaProject.getProject();
+    public void initialize(IPath containerPath, final IJavaProject javaProject) throws CoreException {
+        Central.onWorkspaceInit(new Function<Workspace,Void>() {
+            @Override
+            public Void apply(Workspace a) {
+                try {
+                    Central.getInstance().addModelListener(BndContainerInitializer.this);
```

Comment:
I think you need two onWorkspaceInit calls. One in the constructor to add the listener and another here to do the project initialization.